libpod/config/config.go

```diff
@@ -456,44 +450,45 @@ func NewConfig(userConfigPath string) (*Config, error) {
 		return nil, errors.Wrapf(err, "error finding config on system")
 	}
 
-	migrated := false
 	for _, path := range configs {
-		systemConfig, err := readConfigFromFile(path)
+		config, err = readConfigFromFile(path, config)
 		if err != nil {
 			return nil, errors.Wrapf(err, "error reading system config %q", path)
 		}
-		// Handle CGroups v2 configuration migration.
-		// Migrate only the first config, and do it before
-		// merging.
-		if !migrated {
-			if err := cgroupV2Check(path, systemConfig); err != nil {
-				return nil, errors.Wrapf(err, "error rewriting configuration file %s", userConfigPath)
-			}
-			migrated = true
-		}
-		// Merge the it into the config. Any unset field in config will be
-		// over-written by the systemConfig.
-		if err := config.mergeConfig(systemConfig); err != nil {
-			return nil, errors.Wrapf(err, "error merging system config")
-		}
-		logrus.Debugf("Merged system config %q: %v", path, config)
 	}
 
-	// Finally, create a default config from memory and forcefully merge it into
-	// the config. This way we try to make sure that all fields are properly set
-	// and that user AND system config can partially set.
-	defaultConfig, err := defaultConfigFromMemory()
-	if err != nil {
-		return nil, errors.Wrapf(err, "error generating default config from memory")
+	// First, try to read the user-specified config
+	if userConfigPath != "" {
+		var err error
+		config, err = readConfigFromFile(userConfigPath, config)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error reading user config %q", userConfigPath)
+		}
 	}
 
-	// Check if we need to switch to cgroupfs and logger=file on rootless.
-	defaultConfig.checkCgroupsAndLogger()
-
-	if err := config.mergeConfig(defaultConfig); err != nil {
-		return nil, errors.Wrapf(err, "error merging default config from memory")
+	// Since runc does not currently support cgroupV2
+	// Change to default crun on first running of libpod.conf
+	// TODO Once runc has support for cgroups, this function should be removed.
```

Comment:
I wonder if we should make a backlog jira card to help to remember to remove this.